src/gui/MixerView.cpp

```diff
@@ -185,11 +272,61 @@ int MixerView::addNewChannel()
 
 	updateMixerLine(newChannelIndex);
 
+	updateAutoTrackSortOrder();
 	updateMaxChannelSelector();
 
 	return newChannelIndex;
 }
 
+void MixerView::updateAutoTrackSortOrder()
+{
+
+	Mixer * mix = Engine::mixer();
+	QList<int> *list = new QList<int>();
+
+	// add all non auto track first
+	for( int i = 1; i<m_mixerChannelViews.size(); ++i )
+	{
+		if (!mix->mixerChannel(i)->m_autoTrackLinkModel.value())
+		{
+			list->append(i);
+		}
+	}
+
+	// add auto tracks in the order of the song tracks
+	mix->processAssignedTracks([&, list](Track * track, IntModel * model, MixerChannel * channel)
+	mutable {
+		(void) track;		
+		if (channel == NULL) return;		
+		if (channel->m_autoTrackLinkModel.value())
+		{
+			list->append(model->value());
+		}
+	});
+	return;
+
+
+	// bubblesort here because the list is normally almost ordered
+	int n = list->length();
+	bool swapped = false;
```

Comment:
``swapped`` is not reassigned any value in this do-while loop. Wouldn't this just run once? Unless the intent is something different, I think this can be simplified.

Reply:
The sorting approach is not working out as I thought - therefore the return - but I had no time yet to think again about it.